### PR TITLE
llvm_{6,7}: enable BPF target, fix bcc

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Note sure if other bits benefit from this target as well,
but BCC is useful! O:)

cc #53907

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/llvm/6/llvm.nix [hunk 1]
```diff
@@ -13,9 +13,11 @@
 , fetchpatch
 , debugVersion ? false
 , enableManpages ? false
-# Mesa requires AMDGPU target
-, enableTargets ? [ stdenv.hostPlatform stdenv.targetPlatform "AMDGPU" ]
 , enableSharedLibraries ? true
+# Mesa requires AMDGPU target
+# BPF is used by bcc
+, enableTargets ? [ stdenv.hostPlatform stdenv.targetPlatform "AMDGPU" "BPF" ]
+, enablePFM ? !stdenv.isDarwin
```
Comment (Ericson2314) [inline pkgs/development/compilers/llvm/6/llvm.nix]: Is this used? Looks like a copy-pasta.

**Comment by dtzWill** [comment]
On Sun, 13 Jan 2019 19:15:13 -0800, John Ericson <notifications@github.com> wrote:
> Ericson2314 commented on this pull request.
> 
> 
> 
> >  , enableSharedLibraries ? true
> +# Mesa requires AMDGPU target
> +# BPF is used by bcc
> +, enableTargets ? [ stdenv.hostPlatform stdenv.targetPlatform "AMDGPU" "BPF" ]
> +, enablePFM ? !stdenv.isDarwin
> 
> Is this used? Looks like a copy-pasta.

Yikes, good catch ty.  Fixed!

**Comment by Ericson2314** [comment]
And yes, the concept seems fine to me :) though I don't even know what this bcc is!

**Comment by Mic92** [comment]
I am all for `bcc` to work! Great performance/debugging toolbox.

**Pull request merged**